job_executor/model/datastore.py

```diff
@@ -265,84 +268,136 @@ def bump_version(
+                        if dataset.name != dataset_name
+                    ]
+                    new_metadata_datasets.append(Metadata(**released_metadata))
+                if operation in ['ADD', 'CHANGE_DATA']:
+                    self._log(
+                        job_id, 'Renaming data file and updating data_versions'
+                    )
+                    new_data_versions[dataset_name] = (
+                        local_storage.rename_parquet_draft_to_release(
+                            dataset_name, new_version
+                        )
+                    )
+
+            if update_type in ['MINOR', 'MAJOR']:
+                self._log(job_id, 'Writing new data_versions to file')
+                local_storage.write_data_versions(
+                    new_data_versions, new_version
                 )
-        if update_type in ['MINOR', 'MAJOR']:
-            local_storage.write_data_versions(new_data_versions, new_version)
-        new_metadata_all_dict = self.metadata_all_draft.dict(by_alias=True)
-        del new_metadata_all_dict['dataStructures']
-        new_metadata_all_dict['dataStructures'] = [
-            dataset.dict(by_alias=True) for dataset in new_metadata_datasets
-        ]
-        local_storage.write_metadata_all(
-            new_metadata_all_dict, new_version
-        )
-        self.metadata_all_latest = MetadataAll(
-            **local_storage.get_metadata_all(new_version)
-        )
-        self.latest_version_number = new_version
-        self.metadata_all_draft.remove_all()
-        for metadata in self.metadata_all_latest:
-            self.metadata_all_draft.add(metadata)
-        for draft in self.draft_version:
-            self.metadata_all_draft.remove(draft.name)
-            if draft.operation == 'REMOVE':
-                continue
-            else:
-                self.metadata_all_draft.add(
-                    Metadata(**local_storage.get_metadata(draft.name, 'DRAFT'))
-                )
-        job_service.update_job_status(job_id, 'completed')
-        local_storage.delete_temporary_backup()
+
+            self._log(job_id, 'Writing new metadata_all to file')
+            new_metadata_all_dict = self.metadata_all_draft.dict(by_alias=True)
+            del new_metadata_all_dict['dataStructures']
+            new_metadata_all_dict['dataStructures'] = [
+                dataset.dict(by_alias=True)
+                for dataset in new_metadata_datasets
+            ]
+            local_storage.write_metadata_all(
+                new_metadata_all_dict, new_version
+            )
+            self.metadata_all_latest = MetadataAll(
+                **local_storage.get_metadata_all(new_version)
+            )
+            self.latest_version_number = new_version
+
+            self._log(job_id, 'Rebuilding metadata_all_DRAFT')
+            self.metadata_all_draft.remove_all()
+            for metadata in self.metadata_all_latest:
```

Comment:
All this logic could down to line 395 be well hidden in MetadataAllDraft class, e.g. `self.metadata_all_draft.rebuild(self.metadata_all_latest, self.draft_version)`

Reply:
I agree 👍🏻 I would like to wait until we remove the /metadata folder though since this will be rewritten slightly in that case (use previous metadata_all__DRAFT to generate the new one instead of the file system)

Reply:
I fixed it anyway it was more straight forward than I assumed 👍🏻